src/string.rs

```diff
@@ -252,6 +327,38 @@ impl<const N: usize> String<N> {
         &mut self.vec
     }
 
+    /// Returns a mutable reference to the contents of this `String`.
+    ///
+    /// # Safety
+    ///
+    /// This function is unsafe because it does not check that the bytes passed
+    /// to it are valid UTF-8. If this constraint is violated, it may cause
+    /// memory unsafety issues with future users of the `String`, as the rest of
+    /// the library assumes that `String`s are valid UTF-8.
+    ///
+    /// # Examples
+    ///
+    /// Basic usage:
+    ///
+    /// ```
+    /// use heapless::String;
+    ///
+    /// let mut s: String<8> = String::try_from("hello")?;
+    ///
+    /// unsafe {
+    ///     let vec = s.as_mut_vec_view();
+    ///     assert_eq!(&b"hello", &vec);
```

Comment:
```suggestion
    ///     assert_eq!(&vec, &b"hello");
```

Reply:
Done in 6f4bc0698072e178d72cc29e82398d8a69e530b1